executable-spec/src/Cardano/Ledger/Spec/STS/Update/Data.hs

```diff
@@ -1,29 +1,130 @@
+{-# LANGUAGE DeriveAnyClass #-}
 {-# LANGUAGE DeriveGeneric #-}
 {-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE TypeSynonymInstances #-}
+{-# LANGUAGE FlexibleInstances #-}
 
 module Cardano.Ledger.Spec.STS.Update.Data where
 
-import Data.Word (Word64)
+--import Data.Word (Word64)
 import           Data.Monoid.Generic (GenericMonoid (GenericMonoid),
                      GenericSemigroup (GenericSemigroup))
 import           Data.Set (Set)
 import           GHC.Generics (Generic)
-
+import           Data.Text (Text)
+import           Numeric.Natural (Natural)
+import           Data.Hashable (Hashable)
+import qualified Data.Hashable as H
 import qualified Ledger.Core as Core
+import           Data.Map.Strict (Map)
+
+
+-- |A unique ID of a software update
+newtype UpId = UpId { getUpId :: Integer --Word64
+                    }
+  deriving (Show, Eq, Ord, Hashable, Generic)
+
+instance Core.HasHash (UpId) where
```

Comment:
Parenthesis aren't needed. You could run `hlint` on the source files to detect these BTW ;)

Reply:
OK. I will remove the parenthesis. Also, I will install HLint and try to use it from now on.